gamification-twitter-services/src/main/java/io/meeds/gamification/twitter/service/impl/TwitterAccountServiceImpl.java

```diff
@@ -130,7 +126,10 @@ public void addTwitterAccount(String twitterUsername, String currentUser) throws
     if (!Utils.isRewardingManager(currentUser)) {
       throw new IllegalAccessException("The user is not authorized to add a twitter watched account");
     }
-    RemoteTwitterAccount remoteTwitterAccount = twitterConsumerStorage.retrieveTwitterAccount(twitterUsername,
+    if (twitterAccountStorage.countTwitterAccounts() >= 2) {
+      throw new IllegalAccessException("The maximum number of watched twitter accounts has been reached");
```

Comment:
```suggestion
      throw new IllegalStateException("The maximum number of watched twitter accounts has been reached");
```

in REST, it should return 400 Bad request. It's not about authorization but API limit.